yt/frontends/ramses/field_handlers.py

```diff
@@ -448,7 +448,12 @@ def read_rhs(cast):
             # Read nRTvar, nions, ngroups, iions
             for i in range(4):
                 read_rhs(int)
-            f.readline()
+            # Try to read rtprecision
+            try:
+		read_rhs(int)
+		f.readline()
+            except Exception:
+                pass    
```

Comment:
I am a bit concerned about these lines, as the `Exception` is likely to be raised after the line has been read (see line 443). As a consequence, if the code fails to parse the input as `rtprecision`, then you should rewind the line.

Also, why do you read an extra line after calling `read_rhs`? This is only required if the immediate line after `rtprecision` is blank.
I would suggest

```suggestion
            try:
                ipos = f.tell()
                read_rhs(int)
            except Exception:
                f.seek(ipos)
```

Reply:
Hi Corentin,

Thanks for looking at this. I wanted to make a small push before I start pushing other modifications I've done to my yT that might come useful to other people.

Certainly, I can share a fragment of one of the new files here. The top of a file with the new format looks as follows:
`nRTvar      =         12
nIons       =          3
nGroups     =          3
iIons       =         10
rtprecision =          8
 
X_fraction  =  0.759999990463257E+00
Y_fraction  =  0.239999994635582E+00
 
unit_np     =  0.462706068183299E+04
unit_pf     =  0.693930649464614E+13`

The only difference with the previous format is the addition of an extra line containing an integer for rtprecision. As you mention in the new format, the code will read the new line and then the following one, which is an empty line, so it does work in that instance. Alternatively, in the old version, the exception will be raised when trying to split the empty line, thus there is no need for a second readline. I thought this a simple way to make the update, as it works with both formats, and should be easy to update in the future. I have also tested that it correctly works with both type of files.

Reply:
Oh I see, my bad! I hadn't seen you replaced `f.readline()` by your code snippet.